client/src/components/PlaceLayerModal.js

```diff
@@ -24,15 +27,42 @@ const { LayerTypes } = PlaceLayerUtils;
 
 const Tabs = {
   url: 0,
-  file: 1
+  file: 1,
+  mapLibrary: 3
 };
 
 const PlaceLayerModal: AbstractComponent<any> = (props: Props) => {
   const [loading, setLoading] = useState(false);
   const [tab, setTab] = useState(props.item.content ? Tabs.file : Tabs.url);
+  const [geoLayers, setGeoLayers] = useState([]);
+  const { project } = useContext(ProjectContext);
 
   const { t } = useTranslation();
 
+  /**
+   * Fetch and set the list of georeferenced layers
+   * from the map_library_url, if valid.
+   */
+  useEffect(() => {
```

Comment:
What are your thoughts on doing this on the server? Currently the "Map Library" is just some JSON sitting in an S3 bucket, but at some point it's going to live elsewhere and be editable by the end user. Moving the requests for the map library to the server would eliminate CORS errors and allow us to authenticate if need be. We do something similar with [authorities](https://github.com/performant-software/core-data-connector/tree/9a435434f9b0d9abc6f1f7bd7dfd4c8bdc006cee/app/services/core_data_connector/authority).

Reply:
Makes sense to me, especially if we already have this as a pattern! I'll go ahead and implement it server-side.